### ISSUE TITLE
Refusing new connections after reload the nginx

### ISSUE THREAD

**Issue by gugansankar** [arut/nginx-rtmp-module#580]
Thanks for this Module.

Everything is working fine with this Module. But let me describe the problem which I'm facing now.

1. After I did small changes on my rtmp config, I just reload the nginx service. Then, its started to refuse the new connections. 

  It will back normal only if I stop and restart ed the publisher [ FMLE]. 

Can you help me on this 

Thanks.

**Comment by cccaochao** [comment]
Are you sure new worker had started after reload?The old worker will refuse the new connections.

**Comment by gugansankar** [comment]
@cccaochao , Yes, New worker process got started. But old one was not shutting down. 

Before reload the nginx ,

root      9676     1  0 00:32 ?        00:00:00 nginx: master process /usr/sbin/nginx -c /etc/conf/nginx/nginx.conf
nginx     9707  9676  0 00:34 ?        00:00:00 nginx: worker process 

After reload the Nginx ,
root      9676     1  0 00:32 ?        00:00:00 nginx: master process /usr/sbin/nginx -c /etc/conf/nginx/nginx.conf
nginx     9707  9676  0 00:34 ?        00:00:00 nginx: worker process is shutting down       
nginx     9845  9676  0 00:43 ?        00:00:00 nginx: worker process 


Error in Nginx , [ After reload/restart nginx, the below message writing frequently ].
2015/03/17 00:35:31 [debug] 9678#0: epoll: fd:21 ev:0001 d:00007F72A82755F8
2015/03/17 00:35:31 [debug] 9678#0: *3 recv: fd:21 165 of 175
2015/03/17 00:35:31 [debug] 9678#0: *3 RTMP bheader fmt=1 csid=4
2015/03/17 00:35:31 [debug] 9678#0: *3 RTMP mheader fmt=1 audio (8) time=135805+16 mlen=157 len=0 msid=1
2015/03/17 00:35:31 [debug] 9678#0: *3 RTMP recv audio (8) csid=4 timestamp=135821 mlen=157 msid=1 nbufs=1
2015/03/17 00:35:31 [debug] 9678#0: *3 nhandlers: 5
2015/03/17 00:35:31 [debug] 9678#0: *3 calling handler 0
2015/03/17 00:35:31 [debug] 9678#0: *3 calling handler 1
2015/03/17 00:35:31 [debug] 9678#0: *3 calling handler 2
2015/03/17 00:35:31 [debug] 9678#0: *3 live: audio packet timestamp=135821
2015/03/17 00:35:31 [debug] 9678#0: *3 RTMP prep audio (8) fmt=1 csid=6 timestamp=26 mlen=157 msid=1 nbufs=1
2015/03/17 00:35:31 [debug] 9678#0: *3 calling handler 3
2015/03/17 00:35:31 [debug] 9678#0: *3 calling handler 4
2015/03/17 00:35:31 [debug] 9678#0: *3 recv: fd:21 -1 of 175
2015/03/17 00:35:31 [debug] 9678#0: *3 recv() not ready (11: Resource temporarily unavailable)
2015/03/17 00:35:31 [debug] 9678#0: timer delta: 1
2015/03/17 00:35:31 [debug] 9678#0: posted events 0000000000000000
2015/03/17 00:35:31 [debug] 9678#0: worker cycle
2015/03/17 00:35:31 [debug] 9678#0: epoll timer: 40354

Can you pls check this 

Thanks

**Comment by cccaochao** [comment]
I think because  you not close the publisher and the old worker can't shut down . so ,the new connetion at the new worker that have no this stream.

**Comment by gugansankar** [comment]
@cccaochao , Thanks for your comments. But I'm not clear why do we need to close the publisher.

For example, After changing the Flash Media Server configuration, we just restart the FMS service and continue the process. Not going to do the close the publisher [ FMLE ] connection. 
 But Here also, I did the some process [ did some config changes ]. After nginx reload/restart, video not delivering for new connection.

Please correct me If I am wrongly mentioned anything. And help me to fix this issue

Thanks.

**Comment by cccaochao** [comment]
@gugansankar   nginx will shut down the old worker that no connection and timer. If you don't close the publisher, the old worker always "nginx: worker process is shutting down" . so I think nginx's reload is not suitable for these long connection.

**Comment by gugansankar** [comment]
@cccaochao , Do you mean, bug with this module ?. Are you using this module and don't have such type of issue ?

**Comment by misiek08** [comment]
This module is not supporting reload. It's not a bug, just not implemented
and that's all.
17 mar 2015 08:08 "gugansankar" <notifications@github.com> napisał(a):

**Comment by gugansankar** [comment]
Any suggestion for configuration take effect after modification ?.  Every time going to restart the publisher  FMLE ] is not a feasible one. I'm wondering some way to solve this. Can you give some alternate work around solution for this.?

**Comment by cccaochao** [comment]
@gugansankar  Yes , i meet this problem,I change many codes  and only reload some directive. You only kill nginx and restart. If you want this function,I recommend one rtmp server : 
simple rtmp server 
https://github.com/winlinvip/simple-rtmp-server
